src/main/java/school/hei/patrimoine/visualisation/swing/ihm/google/GoogleDocsLinkVerfierScreen.java

```diff
@@ -0,0 +1,214 @@
+package school.hei.patrimoine.visualisation.swing.ihm.google;
+
+import static java.awt.BorderLayout.CENTER;
+import static java.awt.Font.BOLD;
+import static java.awt.Font.PLAIN;
+import static java.awt.GridBagConstraints.HORIZONTAL;
+import static javax.swing.SwingConstants.LEFT;
+import static javax.swing.SwingUtilities.invokeLater;
+import static javax.swing.WindowConstants.EXIT_ON_CLOSE;
+
+import java.awt.*;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import javax.swing.*;
+import lombok.extern.slf4j.Slf4j;
+import school.hei.patrimoine.compiler.ClassNameExtractor;
+import school.hei.patrimoine.compiler.PatrimoineCompiler;
+import school.hei.patrimoine.google.GoogleApi;
+import school.hei.patrimoine.google.GoogleApi.GoogleAuthenticationDetails;
+import school.hei.patrimoine.google.GoogleDocsLinkIdParser;
+import school.hei.patrimoine.modele.Patrimoine;
+import school.hei.patrimoine.visualisation.swing.ihm.MainIHM;
+import school.hei.patrimoine.visualisation.swing.ihm.google.modele.NamedSnippet;
+import school.hei.patrimoine.visualisation.swing.ihm.google.modele.NamedString;
+import school.hei.patrimoine.visualisation.swing.ihm.google.modele.NamedURL;
+
+@Slf4j
+public class GoogleDocsLinkVerfierScreen {
+  private final JFrame inputFrame;
+  private final JPanel inputPanel;
+  private final List<JTextField> inputFields;
+  private final GoogleDocsLinkIdInputVerifier linkIdInputVerifier =
+      new GoogleDocsLinkIdInputVerifier();
+  private final GoogleDocsLinkIdParser linkIdParser = new GoogleDocsLinkIdParser();
+  private final GoogleApi googleApi;
+  private final GoogleAuthenticationDetails authDetails;
+
+  public GoogleDocsLinkVerfierScreen(
+      GoogleApi googleApi, GoogleAuthenticationDetails authDetails, List<NamedString> linksData) {
```

Comment:
store linksData in a private final attribute and use it instead of passing it down to all your methods

Reply:
done